Source/Core/Common/Logging/Log.h

```diff
@@ -76,13 +78,17 @@ enum LOG_LEVELS
 static const char LOG_LEVEL_TO_CHAR[7] = "-NEWID";
 
 void GenericLogFmtImpl(LOG_LEVELS level, LOG_TYPE type, const char* file, int line,
-                       std::string_view format, const fmt::format_args& args);
+                       fmt::string_view format, const fmt::format_args& args);
 
-template <typename... Args>
-void GenericLogFmt(LOG_LEVELS level, LOG_TYPE type, const char* file, int line,
-                   std::string_view format, const Args&... args)
+template <std::size_t NumFields, typename S, typename... Args>
+void GenericLogFmt(LOG_LEVELS level, LOG_TYPE type, const char* file, int line, const S& format,
+                   const Args&... args)
 {
-  GenericLogFmtImpl(level, type, file, line, format, fmt::make_format_args(args...));
+  static_assert(
+      NumFields == sizeof...(args),
+      "Unexpected number of replacement fields in format string; did you pass too many arguments?");
```

Comment:
Should this message perhaps go both ways, like "[...] did you pass too few or too many arguments"?